docs/topics/testing/advanced.txt

```diff
@@ -781,6 +781,41 @@ utility methods in the ``django.test.utils`` module.
     configuration that need to be reversed. It's the return value of the
     :meth:`setup_databases` method.
 
+.. function:: isolate_apps(*app_labels, attr_name=None, kwarg_name=None)
+
+    Register the models defined within a wrapped context into their own
+    isolated app registry. This functionality is useful when creating a
+    Django model for a test case. An isolated registry avoids mixing
+    concerns as well as edge cases involving name collision.
+
+    The app labels which the isolated registry should contain must be
+    passed as individual arguments. And the invocation can be either a
+    decorator or a context manager. For example::
+
+        class MyModelTests(SimpleTestCase):
+
+            @isolate_apps("app_label")
+            def test_model_definition(self):
+                class TestModel(models.Model):
+                    pass
+                ...
+
+    Or::
+
+        with isolate_apps("app_label"):
+            class TestModel(models.Model):
+                pass
+            ...
+
+    See also, :ref:`isolating-model-registration`.
```

Comment:
I'm not sure about the main django docs linking to the contributing docs. It could be confusing for users to find themselves with loads of other information about "how you should be doing things" that's more specific to django's code base. Better to move any useful information from that section here, and leave the contributing docs with just a pointer to `isolate_apps`.

Reply:
👍